services/data-service/src/main/java/at/escapedoom/data/service/SceneService.java

```
@@ -118,14 +133,29 @@ public DeleteLevelResponseDTO deleteScene(String id) {
 
         Scene scene = optionalScene.get();
 
-        Level parentLevel = levelRepository.findAll().stream().filter(level -> level.getScenes().contains(scene))
-                .findFirst().orElse(null);
+        Optional<Level> parentLevelOptional = levelRepository.findAll().stream()
+                .filter(level -> level.getScenes() != null && level.getScenes().contains(scene)).findFirst();
+
+        Level parentLevel = parentLevelOptional.orElse(null);
 
         if (parentLevel != null) {
+            try {
+                parentLevel.getScenes().remove(scene);
+                levelRepository.saveAndFlush(parentLevel);
+            } catch (Exception e) {
+                log.error("Failed to delete scene from level", e);
+            }
             parentLevel.getScenes().remove(scene);
             levelRepository.saveAndFlush(parentLevel);
         }
 
+        try {
+            scene.setLevel(null);
+            sceneRepository.delete(scene);
+            sceneRepository.flush();
+        } catch (Exception e) {
+            log.error("Failed to delete scene", e);
```

Warum log sollten wir nicht thronen das es der User mitbekommt

---

Außerdem ist .delete() soweit ich weiß ein silent fail - Aka das catch wird niemals aufgrund eines fehlgeschlagenen deletes triggeren oder ?
https://github.com/spring-projects/spring-data-commons/issues/2651